checks/raw/security_policy.go

```diff
@@ -112,3 +151,114 @@ func isSecurityPolicyFilename(name string) bool {
 		strings.EqualFold(name, "doc/security.rst") ||
 		strings.EqualFold(name, "docs/security.rst")
 }
+
+var checkSecurityPolicyFileContent fileparser.DoWhileTrueOnFileContent = func(path string, content []byte,
+	args ...interface{},
+) (bool, error) {
+	if len(args) != 2 {
+		return false, fmt.Errorf(
+			"checkSecurityPolicyFileContent requires exactly two arguments: %w", errInvalidArgLength)
+	}
+	pfiles, ok := args[0].(*checker.File)
+	if !ok {
+		return false, fmt.Errorf(
+			"checkSecurityPolicyFileContent requires argument of type *checker.File: %w", errInvalidArgType)
+	}
+	pinfo, ok := args[1].(*[]checker.SecurityPolicyInformation)
+	if !ok {
+		return false, fmt.Errorf(
+			"%s requires argument of type *[]checker.SecurityPolicyInformation: %w",
+			"checkSecurityPolicyFileContent", errInvalidArgType)
+	}
+
+	if len(content) == 0 {
+		// perhaps there are more policy files somewhere else,
+		// keep looking (true)
+		return true, nil
+	}
+
+	if pfiles != nil && (*pinfo) != nil {
+		// preserve file type
+		tempType := pfiles.Type
+		policyHits := collectPolicyHits(content)
+
+		*pfiles = checker.File{
+			Path:   path,
+			Type:   tempType,
+			Offset: checker.OffsetDefault,
+			// convey the length/amount of content using
+			// the EndOffset as the len to EOF used in eval
+			EndOffset: uint(len(content)),
```

Comment:
I think it's fine to add a `uint * size` in the File structure. EndOffset is for code snippet. We use in in SARIF, for example.

Reply:
Added `FileSize` to `File struct` to hold this information thereby allowing removal of `SecurityContentLength` which I think nicely cleaned up that part of the code.